src/json_output.rs

```diff
@@ -97,24 +96,33 @@ impl JsonOutput {
         }
     }
 
-    fn insert_node_without_attributes(&mut self, _: &XmlElement, name: &str) -> Result<()> {
+    fn insert_node_without_attributes(
+        &mut self,
+        _: &XmlElement,
+        name: &str,
+    ) -> SerializationResult<()> {
         trace!("insert_node_without_attributes");
         self.stack.push(name.to_owned());
 
         let container =
             self.get_current_parent()
                 .as_object_mut()
-                .context(err::JsonStructureError {
+                .ok_or(SerializationError::JsonStructureError {
```

Comment:
`ok_or_else` here as well (and in the rest of them)

Reply:
👍